ls1intum/Artemis
# `Exam mode`: Update exam scores table according to filter configuration

### Checklist
#### General


#### Client

### Motivation and Context


The exam scores page provides two filter options that can be selected separately: Include only submitted exams and include only exams with at least one submission. These filter options are currently only applied to the charts on this page, but not on the table right below the filter options. This is changed by this PR.
### Description

I reworked the structure containing the values of the table and added the necessary variants of the statistical values that are shown by the table. Finally, I added methods that update the table state according to the current filter option.
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam with participation
**Before deploying this PR**: Inspect the table of the exam scores and take a screenshot. This might help you to confirm that the same data is displayed per default.

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to the scores page of the corresponding exam
4. Check the table content. Make sure that still the same data is displayed
5. Play around with the filters. Make sure that the table content changes if appropriate. The easiest cases would be the percentage of passed or submitted exams if you choose a filter.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





| File name | Branch coverage (%) | Line coverage (%) |
| -- | -- | -- |
| `exam-scores.component.ts` | 83.72 | 97.62 |

### Screenshots


Default state:
![image](https://user-images.githubusercontent.com/80622272/156559814-f89a5dae-5cbd-42ba-a762-976c89c3f7f0.png)

Include only submitted exams:
![image](https://user-images.githubusercontent.com/80622272/156559887-b8c5cd3e-bdb2-4695-9067-afbec37135f3.png)

Include only non empty submissions:
![image](https://user-images.githubusercontent.com/80622272/156559956-60486156-b4ba-4527-a47b-aa7eb1dfee5a.png)

Both filters active:
![image](https://user-images.githubusercontent.com/80622272/156560046-cdae4d2a-493d-4a6f-810d-515be2afcb04.png)
